src/main/java/nl/uu/cs/ape/Main.java

```diff
@@ -27,12 +31,119 @@ public class Main {
      * Interface (CLI).
      *
-     * @param args APE expects at most one (1) argument: The absolute or relative
-     *             path to the configuration file.
+     * @param args APE expects the first argument to specify the method to be
+     *             executed.
+     *             The rest of the arguments are provided depending on the method.
      */
     public static void main(String[] args) {
+
+        String method = args[0];
+        switch (method) {
+            case "synthesis":
+                executeSynthesis(ArrayUtils.remove(args, 0));
+                break;
+            case "convert-tools":
+                convertBioToolsAnnotations(ArrayUtils.remove(args, 0));
+                break;
+            case "pull-a-tool":
+                pullATool(ArrayUtils.remove(args, 0));
+                break;
+            case "bio.tools":
+                try {
+                    BioToolsAPI.getAndSaveFullBioTools("./tools.json");
+                } catch (IOException e) {
+                    log.error("Error in fetching the tools from bio.tools.");
+                }
+                break;
+            default:
+                if (args.length == 0 || args.length == 1) {
+                    executeSynthesis(args);
+                } else {
+                    log.error("Invalid method provided.");
+                }
+        }
+
+    }
+
+    private static void pullATool(String[] args) {
+        if (args.length != 1) {
+            log.error("Error: pull-a-tool method expects biotoolsID as the only additional argument.");
+            return;
+        }
+
+        String biotoolsID = args[0];
+        if (biotoolsID.isEmpty()) {
+            log.error("Error: biotoolsID should be provided as an additional argument.");
+            return;
+        }
+
+        try {
+            JSONObject tool = BioToolsAPI.getAndConvertToolList(List.of(biotoolsID));
+            APEFiles.write2file(tool.toString(4), new File("./" + biotoolsID + ".json"), false);
+
+            JSONObject toolAnnotation = tool.getJSONArray("functions").getJSONObject(0);
+            APE apeFramework = new APE(
+                    "https://raw.githubusercontent.com/Workflomics/tools-and-domains/refs/heads/main/domains/proteomics/config.json");
+
+            Module cometModule = apeFramework.getDomainSetup()
+                    .updateModuleFromJson(toolAnnotation).get();
+
+            ToolCWLCreator toolCWLCreator = new ToolCWLCreator(cometModule);
+
+            APEFiles.write2file(toolCWLCreator.generate(), new File("./" + biotoolsID + ".cwl"), false);
+
+        } catch (IOException e) {
+            log.error("Error in fetching the tool from bio.tools.");
+        } catch (OWLOntologyCreationException e) {
+            log.error(
+                    "Error in setting up the APE framework from the 'https://raw.githubusercontent.com/Workflomics/tools-and-domains/refs/heads/main/domains/proteomics/config.json' configuration.");
```

Comment:
Perhaps it would be a good idea to set the URL as variable and use it in lines 86 and 99 to improve maintainability.

Reply:
Good point, I updated it locally, and I will push it with the new commit. This may not be the permanent solution because it hardcodes the domain that can be used with the feature, but it should be good enough for now.

Reply:
Yeah, but currently, the domain is fixed. So I don't see a problem here for now.

Reply:
I agree